core/src/test/java/org/apache/calcite/test/SqlValidatorTest.java

```diff
@@ -6443,6 +6443,26 @@ public void _testGroupExpressionEquivalenceParams() {
         + " count(empno + deptno) OVER (partition by empno + deptno + 1)\n"
         + " from emp group by empno + deptno");
 
+    // expression is NOT OK (AS clause)
+    sql("select sum(max(empno))\n"
+        + " OVER (order by ^deptno^ ROWS 2 PRECEDING) AS sumEmpNo\n"
+        + " from emp")
+        .fails("Expression 'DEPTNO' is not being grouped");
+    sql("select sum(max(empno)) OVER w AS sumEmpNo\n"
+        + " from emp\n"
+        + " window w AS (order by ^deptno^ ROWS 2 PRECEDING)")
+        .fails("Expression 'DEPTNO' is not being grouped");
+    sql("select avg(^sal^) OVER () AS avgSal,"
+        + " avg(count(empno)) OVER (partition by 1) AS avgEmpNo\n"
+        + " from emp")
+        .fails("Expression 'SAL' is not being grouped");
+    sql("select count(deptno, deptno + 1, ^empno^) OVER () AS cntDeptEmp\n"
```

Comment:
According to https://calcite.apache.org/docs/reference.html
COUNT( [ DISTINCT ] value [, value]* ) - Returns the number of input rows for which value is not null (wholly not null if value is composite)